### PR TITLE
fix: getX queries for mongo and pagination

### PR DESCRIPTION
**Pull request by wtrocki**
Fixing small issues with mongodb querying

## Verification
1. Run mongodb template
2. Execute queries in order

```graphql

mutation create {
  createNote(input: { id: "", title: "test", description: "" }) {
    id
    title
  }
}

query getNote {
  getNote(id: "5ec65b464edd042b5c862286") {
    id
  }
}

query query2 {
  findNotes(
    filter: { description: {eq:"" } }
    page: { offset: 0, limit: 40 }
    orderBy: { field: "title", order: ASC }
  ) {
    items {
      id
      title
      description
    }
  }
}


```

### CODE REVIEW

**Comment by wtrocki** [comment]
Build failing because of keycloak auth package

File: packages/graphback-runtime-knex/src/KnexDBDataProvider.ts [hunk 1]
```diff
@@ -123,20 +123,19 @@ export class KnexDBDataProvider<Type = any, GraphbackContext = any> implements G
...
     }
 
-    if (offset) {
-      query = query.offset(offset)
-    }
-
-    if (limit <= 0) {
-      throw new Error("Please use a limit of greater than 0 in queries")
+    if (limit <= 1) {
+      throw new Error("Invalid limit value. Please use a limit of greater than 1 in queries")
```
Comment (craicoverflow) [inline packages/graphback-runtime-knex/src/KnexDBDataProvider.ts:130]: A limit of 1 should be allowed. Someone wants to perform a filter to get all tasks incomplete but they only need the first one.

File: packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts [hunk 2]
```diff
@@ -170,23 +171,25 @@ export class MongoDBDataProvider<Type = any, GraphbackContext = any> implements
...
-        // If no offset is supplied
-        page.offset = defaultOffset
-      } else {
-        if (page.offset < 0) {
-          throw new Error("Please use an offset of greater than or equal to 0 in queries")
-        }
-      }
-      query = query.skip(page.offset)
+    if (!page) {
+      return query.toArray();
```
Comment (craicoverflow) [inline packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts:175]: Should this just return the `query` object so people can perform pagination before they are ready to convert to a list.

File: packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts [hunk 3]
```diff
@@ -170,23 +171,25 @@ export class MongoDBDataProvider<Type = any, GraphbackContext = any> implements
   }
 
   private usePage(query: Cursor<any>, page?: GraphbackPage, defaultLimit: number = 10, defaultOffset: number = 0) {
```
Comment (craicoverflow) [inline packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts:173]: default values can be removed

Comment (craicoverflow) [inline packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts:173]: default values can be removed

File: packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts [hunk 4]
```diff
@@ -170,23 +171,25 @@ export class MongoDBDataProvider<Type = any, GraphbackContext = any> implements
...
-        }
-        query = query.limit(page.limit)
-      }
+    const { offset, limit } = page
+
+    if (offset < 0) {
+      throw new Error("Invalid offset value. Please use an offset of greater than or equal to 0 in queries")
+    }
+
+    if (limit <= 1) {
```
Comment (craicoverflow) [inline packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts:184]: Limit should allow 1

**Pull request merged**